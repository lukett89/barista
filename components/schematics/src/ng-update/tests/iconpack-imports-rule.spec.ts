/**
 * @license
 * Copyright 2019 Dynatrace LLC
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { createTestCaseSetup } from '../../testing';

export const migrationCollection = require.resolve('../../migration.json');

describe('v5 dynatrace angular components imports', () => {
  it('should migrate root imports correctly', async () => {
    const { runFixers, appTree, removeTempDir } = await createTestCaseSetup(
      'update-5.0.0',
      migrationCollection,
      [require.resolve('./dt-iconpack-imports.fixture')],
    );

    if (runFixers) {
      await runFixers();
    }

    expect(
      appTree.readContent(
        'projects/lib-testing/src/tests/dt-iconpack-imports.ts',
      ),
    ).toMatchSnapshot();

    removeTempDir();
  });
});
