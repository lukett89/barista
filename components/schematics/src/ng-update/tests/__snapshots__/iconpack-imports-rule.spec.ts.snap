// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v5 dynatrace angular components imports should migrate root imports correctly 1`] = `
"import { DtIconRegistry } from '@dynatrace/barista-icons';
import * as IconPack from '@dynatrace/barista-icons';
"
`;
